docs/text-predicates.rst

```diff
@@ -0,0 +1,215 @@
+==========================
+JanusGraph Text Predicates
+==========================
+
+JanusGraph provides APIs to query the data using Text Predicates. The types of Text Predicates available
+for querying depends on the Indexing backend used.
+Hence, be careful with using the Text Predicates, and refer to original JanusGraph docs_.
+
+.. _docs: https://docs.janusgraph.org/latest/
+
+
+----------------------------
+Available Text Predicates
+----------------------------
+
+The Text Predicates are broadly divided into 2 categories as:
+    - Text search predicates which match against the individual words inside a text string after it has been tokenized.
+        These predicates are not case sensitive.
+
+        - TextContains_
+        - TextContainsFuzzy_
+        - TextContainsPrefix_
+        - TextContainsRegex_
+
+    - String search predicates which match against the entire string value
+
+        - TextFuzzy_
+        - TextPrefix_
+        - TextRegex_
+
+.. note::
+    All the bellow examples are based on Graph of the Gods graph which comes pre-packaged with JanusGraph
+
+^^^^^^^^^^^^^^^
+TextContains
+^^^^^^^^^^^^^^^
+
+This Predicate is part of *Text* package.
+TextContains is used to query for partial matches in values of properties.
+The query is to find all edges where ``reason`` contains ``breezes``
+
+
+.. code-block:: python
+
+    from janusgraph_python.core.attribute.text import Text
+
+    # Find all edges which contains the word breezes
+    edges = g.E().has("reason", Text.textContains("breezes")).next()
+    print(edges)
+
+    ==> e[55m-6hc-b2t-3bk][8400-lives->4304]
```

Comment:
Why aren't the properties also printed here? I think that would be helpful to understand what the predicate matched on.